.refsym directive is not supported

From newlib/libgloss build:
```
../../.././libgloss/msp430/crt0.S:58:2: error: unknown directive
 .refsym __crt0_call_init_then_main
 ^
```

> .refsym
This directive instructs assembler to add an undefined reference to the symbol following the directive. The maximum symbol name length is 1023 characters. No relocation is created for this symbol; it will exist purely for pulling in object files from archives. Note that this reloc is not sufficient to prevent garbage collection; use a KEEP() directive in the linker file to preserve such objects.

Some details: http://people.redhat.com/~dj/msp430/size-optimizations.html

--- a/src/llvm/lib/Target/MSP430/AsmParser/MSP430AsmParser.cpp
+++ b/src/llvm/lib/Target/MSP430/AsmParser/MSP430AsmParser.cpp
@@ -49,6 +49,7 @@ class MSP430AsmParser : public MCTargetAsmParser {
                         SMLoc NameLoc, OperandVector &Operands) override;
 
   bool ParseDirective(AsmToken DirectiveID) override;
+  bool ParseDirectiveRefSym(AsmToken DirectiveID);
 
   unsigned validateTargetOperandClass(MCParsedAsmOperand &Op,
                                       unsigned Kind) override;
@@ -404,6 +405,16 @@ bool MSP430AsmParser::ParseInstruction(ParseInstructionInfo &Info,
   return false;
 }
 
+bool MSP430AsmParser::ParseDirectiveRefSym(AsmToken DirectiveID) {
+    StringRef Name;
+    if (getParser().parseIdentifier(Name))
+      return TokError("expected identifier in directive");
+
+    MCSymbol *Sym = getContext().getOrCreateSymbol(Name);
+    getStreamer().EmitSymbolAttribute(Sym, MCSA_Global);
+    return false;
+}
+
 bool MSP430AsmParser::ParseDirective(AsmToken DirectiveID) {
   StringRef IDVal = DirectiveID.getIdentifier();
   if (IDVal.lower() == ".long") {
@@ -412,6 +423,8 @@ bool MSP430AsmParser::ParseDirective(AsmToken DirectiveID) {
     ParseLiteralValues(2, DirectiveID.getLoc());
   } else if (IDVal.lower() == ".byte") {
     ParseLiteralValues(1, DirectiveID.getLoc());
+  } else if (IDVal.lower() == ".refsym") {
+    return ParseDirectiveRefSym(DirectiveID);
   }
   return true;
 }

--- a/src/llvm/test/MC/MSP430/refsym.s
+++ b/src/llvm/test/MC/MSP430/refsym.s
@@ -1,0 +1,14 @@
+# RUN: llvm-mc -filetype=obj -triple=msp430 %s | llvm-readobj -t - | FileCheck %s
+
+foo:
+  .refsym __hook
+
+; CHECK:       Symbol {
+; CHECK:         Name: __hook (30)
+; CHECK-NEXT:    Value: 0x0
+; CHECK-NEXT:    Size: 0
+; CHECK-NEXT:    Binding: Global (0x1)
+; CHECK-NEXT:    Type: None (0x0)
+; CHECK-NEXT:    Other: 0
+; CHECK-NEXT:    Section: Undefined (0x0)
+; CHECK-NEXT:  }

